### PR TITLE
Split ValidationSet setters.

### PR DESCRIPTION
**Pull request by robertpustulka**
Deprecate `ValidationSet::isPresenceRequired()` and `ValidationSet::isEmptyAllowed()` as setters.

The method name for a getter is fine, so only the parameters are deprecated. I'm not sure this is a good way to do it though.

### CODE REVIEW

**Comment by markstory** [comment]
I think the approach is pretty good and results in a similar end state to what we have with other split methods. It would be nice to keep a test for the deprecated behaviour though.

**Comment by robertpustulka** [comment]
@markstory tests added

File: tests/TestCase/Validation/ValidationSetTest.php [hunk 1]
```diff
@@ -191,20 +192,129 @@ public function testCount()
...
+    /**
+     * Test isPresenceRequired deprecated setter
+     *
+     * @group deprecated
+     * @return void
+     */
+    public function testRequirePresenceDeprecated()
+    {
+        $this->deprecated(function () {
+            $set = (new ValidationSet);
```
Comment (dereuromark) [inline tests/TestCase/Validation/ValidationSetTest.php]: Why these bracing styles? Shouldnt it be `= new ValidationSet()`?

Comment (robertpustulka) [reply tests/TestCase/Validation/ValidationSetTest.php]: Copy-pasta :)

Comment (dereuromark) [reply tests/TestCase/Validation/ValidationSetTest.php]: Sounds like a sniff is missing for consistent newable :)

**Comment by markstory** [comment]
Looks great! I will get this merged after I get the migration guide updated.

**Pull request merged**